src/pipecat/services/mcp_service.py

```diff
@@ -0,0 +1,213 @@
+import json
+from typing import Any, Dict, List, Optional, Union
+
+from loguru import logger
+
+from pipecat.adapters.schemas.function_schema import FunctionSchema
+from pipecat.adapters.schemas.tools_schema import ToolsSchema
+
+try:
+    from mcp import ClientSession, StdioServerParameters, types
+    from mcp.client.session import ClientSession
+    from mcp.client.sse import sse_client
+    from mcp.client.stdio import stdio_client
+except ModuleNotFoundError as e:
+    logger.error(f"Exception: {e}")
+    logger.error("In order to use an MCP client, you need to `pip install pipecat-ai[mcp]`.")
+    raise Exception(f"Missing module: {e}")
+
+
+class MCPClient:
```

Comment:
Mmmm, the file is called `mcp_service.py` and the class `MCPClient`. Usually, we use the same name. :thinking:

Reply:
We might want to inherit from `BaseObject` which will allow us to add event handlers, etc.

Reply:
Maybe the name is fine, ignore for now.

Reply:
it is a good question- "mcp" isn't really a service. And "MCPClient" implies the existence of "MCPServer" which doesn't exist... I'll not touch the names for now, but I'm open to changing in one direction or another.
the more I think about it, it feels a little odd to pass in "server_params" to a "client"- but maybe not. maybe it makes sense. now I'm overthinking it... 😂